kaggle/api/kaggle_api_extended.py

```diff
@@ -6,6 +6,11 @@
 from os.path import expanduser, isfile
 from datetime import datetime
 
+try:
+    unicode        # Python 2
+except NameError:
+    unicode = str  # Python 3
+
```

Comment:
The `try/except` seems a bit odd given the fact that `six` is already being installed in `setup.py`.

Is there any particular reason you didn't just import `six` and use [`six.text_type`](https://pythonhosted.org/six/#six.text_type) in the `string` function below?

Reply:
@delirious-lettuce Good to bump into you again on the interwebs... Perhaps your approach is acceptable to the maintainers and is clearly much cleaner but changes the logic in Python 2 which to my mind reads: _if item is unicode then return it, otherwise return str(item)_.  This means that in Python 2 __string(42)__ returns __‘42’__, not __u’42’__.   While __type('42') == type(u'42')__ is True in Python 3, it is False in Python 2.

I hope they find your approach acceptable because then we could just eliminate __string()__ which is currently masking the Python standard library [string](https://docs.python.org/3/library/string.html) module.

Reply:
@cchamberlain ,

Nice to bump into you again as well... but I'm not sure I understand your comment.

> To my mind, the current code reads: if item is unicode then return it, otherwise return str(item). This means that in Python 2 string(42) returns ‘42’, not u’42’. While type('42') == type(u'42') is True in Python 3, it is False in Python 2.

```python
>>> import six
... 
... try:
...     unicode  # Python 2
... except NameError:
...     unicode = str  # Python 3
... 
... 
... def string(item):
...     try:
...         if isinstance(item, unicode):
...             return item
...         else:
...             return str(item)
...     except:
...         return str(item)
... 
... 
... def string_six(item):
...     return item if isinstance(item, six.text_type) else str(item)
... 

# Python 2.7.14 (default, Dec 12 2017, 12:40:29) 
# [GCC 5.4.0 20160609] on linux2
>>> string(42) == string_six(42)
True
>>> string(u'some unicode') == string_six(u'some unicode')
True

# Python 3.6.3 (default, Dec 12 2017, 12:53:45) 
# [GCC 5.4.0 20160609] on linux
>>> string(42) == string_six(42)
True
>>> string(u'some unicode') == string_six(u'some unicode')
True

```
It seems like the same output but you get to remove the `try/except` at the top by simply replacing `unicode` with `six.text_type` inside of `isinstance`. Maybe I am missing something?

Also, I agree with you that the name of this function should be changed to something other than `string`.